### PR TITLE
Write a temporary table to verify write permissions when checking destination JDBC credentials

### LINKED ISSUE #1772: Snowflake check does not verify permissions correctly
## Expected Behavior
I expect the snowflake destination to check upfront whether the credentials I input have the correct permissions. 

## Current Behavior
a snowflake sync is succeeding credentials check but failing sync with the following stack trace 

```
2021-01-21 22:27:16 ERROR (/tmp/workspace/6/0) LineGobbler(voidCall):69 - Exception in thread "main" net.snowflake.client.jdbc.SnowflakeSQLException: SQL compilation error:
2021-01-21 22:27:16 ERROR (/tmp/workspace/6/0) LineGobbler(voidCall):69 - Schema 'AIRBYTE_SCHEMA' already exists, but current role has no privileges on it. If this is unexpected and you cannot resolve this problem, contact your system administrator. ACCOUNTADMIN role may be required to manage the privileges on the object.
2021-01-21 22:27:16 ERROR (/tmp/workspace/6/0) LineGobbler(voidCall):69 - at net.snowflake.client.jdbc.SnowflakeUtil.checkErrorAndThrowExceptionSub(SnowflakeUtil.java:124)
2021-01-21 22:27:16 ERROR (/tmp/workspace/6/0) LineGobbler(voidCall):69 - at net.snowflake.client.jdbc.SnowflakeUtil.checkErrorAndThrowException(SnowflakeUtil.java:64)
2021-01-21 22:27:16 ERROR (/tmp/workspace/6/0) LineGobbler(voidCall):69 - at net.snowflake.client.core.StmtUtil.pollForOutput(StmtUtil.java:434)
2021-01-21 22:27:16 ERROR (/tmp/workspace/6/0) LineGobbler(voidCall):69 - at net.snowflake.client.core.StmtUtil.execute(StmtUtil.java:338)
2021-01-21 22:27:16 ERROR (/tmp/workspace/6/0) LineGobbler(voidCall):69 - at net.snowflake.client.core.SFStatement.executeHelper(SFStatement.java:501)
2021-01-21 22:27:16 ERROR (/tmp/workspace/6/0) LineGobbler(voidCall):69 - at net.snowflake.client.core.SFStatement.executeQueryInternal(SFStatement.java:229)
2021-01-21 22:27:16 ERROR (/tmp/workspace/6/0) LineGobbler(voidCall):69 - at net.snowflake.client.core.SFStatement.executeQuery(SFStatement.java:167)
2021-01-21 22:27:16 ERROR (/tmp/workspace/6/0) LineGobbler(voidCall):69 - at net.snowflake.client.core.SFStatement.execute(SFStatement.java:749)
2021-01-21 22:27:16 ERROR (/tmp/workspace/6/0) LineGobbler(voidCall):69 - at net.snowflake.client.jdbc.SnowflakeStatementV1.executeInternal(SnowflakeStatementV1.java:292)
2021-01-21 22:27:16 ERROR (/tmp/workspace/6/0) LineGobbler(voidCall):69 - at net.snowflake.client.jdbc.SnowflakeStatementV1.execute(SnowflakeStatementV1.java:342)
2021-01-21 22:27:16 ERROR (/tmp/workspace/6/0) LineGobbler(voidCall):69 - at io.airbyte.db.jdbc.JdbcDatabase.lambda$execute$0(JdbcDatabase.java:50)
2021-01-21 22:27:16 ERROR (/tmp/workspace/6/0) LineGobbler(voidCall):69 - at io.airbyte.db.jdbc.DefaultJdbcDatabase.execute(DefaultJdbcDatabase.java:57)
2021-01-21 22:27:16 ERROR (/tmp/workspace/6/0) LineGobbler(voidCall):69 - at io.airbyte.db.jdbc.JdbcDatabase.execute(JdbcDatabase.java:50)
2021-01-21 22:27:16 ERROR (/tmp/workspace/6/0) LineGobbler(voidCall):69 - at io.airbyte.integrations.destination.jdbc.DefaultSqlOperations.createSchemaIfNotExists(DefaultSqlOperations.java:43)
```

## Steps to Reproduce
1. Use a role which doesn't have access credentials to the current schema
1. 💣 

## Severity of the bug for you
Medium

## Airbyte Version
@michel-tricot will update

### PR DESCRIPTION
**Pull request by sherifnada**
## What
Closes #1772 

Enhance the `check` method for JDBC destinations by creating & dropping a temporary table. The "correct" way to do this should be to write DB-specific queries (e.g to check the right metadata tables) that don't incur any side-effects, but getting this in for the upcoming release, then we can revisit. 

## Pre-merge checklist
- [ ] bump connector versions & publish

### CODE REVIEW

File: airbyte-integrations/connectors/destination-jdbc/src/main/java/io/airbyte/integrations/destination/jdbc/AbstractJdbcDestination.java [hunk 1]
```diff
@@ -71,6 +73,14 @@ public AirbyteConnectionStatus check(JsonNode config) {
       // attempt to get metadata from the database as a cheap way of seeing if we can connect.
       database.bufferedResultSetQuery(conn -> conn.getMetaData().getCatalogs(), JdbcUtils::rowToJson);
 
+      // verify we have write permissions on the target schema by creating a table with a random name,
+      // inserting a record, then dropping that table
```
Comment (sherifnada) [inline airbyte-integrations/connectors/destination-jdbc/src/main/java/io/airbyte/integrations/destination/jdbc/AbstractJdbcDestination.java:77]:
```suggestion
      // then dropping that table
```

File: airbyte-integrations/connectors/destination-jdbc/src/main/java/io/airbyte/integrations/destination/jdbc/AbstractJdbcDestination.java [hunk 2]
```diff
@@ -71,6 +72,14 @@ public AirbyteConnectionStatus check(JsonNode config) {
       // attempt to get metadata from the database as a cheap way of seeing if we can connect.
       database.bufferedResultSetQuery(conn -> conn.getMetaData().getCatalogs(), JdbcUtils::rowToJson);
 
+      // verify we have write permissions on the target schema by creating a table with a random name,
+      // then dropping that table
+      String outputSchema = namingResolver.getIdentifier(config.get("schema").asText());
+      String outputTableName = "_airbyte_connection_test_" + UUID.randomUUID().toString().replaceAll("-", "");
+      sqlOperations.createSchemaIfNotExists(database, outputSchema);
+      sqlOperations.createTableIfNotExists(database, outputSchema, outputTableName);
+      sqlOperations.dropTableIfExists(database, outputSchema, outputTableName);
```
Comment (cgardens) [inline airbyte-integrations/connectors/destination-jdbc/src/main/java/io/airbyte/integrations/destination/jdbc/AbstractJdbcDestination.java:81]: don't we need to drop the schema too?

Comment (sherifnada) [reply airbyte-integrations/connectors/destination-jdbc/src/main/java/io/airbyte/integrations/destination/jdbc/AbstractJdbcDestination.java:81]: We wouldn't wnat to drop the schema if it already existed

Comment (cgardens) [reply airbyte-integrations/connectors/destination-jdbc/src/main/java/io/airbyte/integrations/destination/jdbc/AbstractJdbcDestination.java:81]: ignore this. i thought this was a test schema.

**Review by michel-tricot — APPROVED** [review]
Cool!

**Review by cgardens — APPROVED** [review]
i am really not a big fan of there being a side effect to a check connection. talked through this with @sherifnada, i'm convinced that there's not a great way of getting this information out of the metadata provided by jdbc. assuming that we don't want to write custom code for each db, this seems like the only way we can verify that the destination will be able to create tables in the target schema. i think i'm convinced that this is something worth checking at check connection and will generally make a user's life better.

**Pull request merged**